### PR TITLE
Replaced http-codes.js with http-reasons

### PR DESCRIPTION
**Pull request by kunagpal**

### CODE REVIEW

**Comment by shamasis** [comment]
Module usage does not match and tests needed to address current failing scenario

**Comment by kunagpal** [comment]
Yes, I realized that. Will be adding tests in the next commit.

**Comment by czardoz** [comment]
@kunagpal Was this change even tested out? The `http-reasons` module exports a `lookup` function. It's not an object you can just read properties from. :confused:

File: test/integration-legacy/runner-spec.test.js [hunk 1]
```diff
@@ -250,6 +251,7 @@ describe('Runner', function () {
                             expect(cursor.ref).to.eql(runStore.ref);
 
                             expect(response.code).to.be(200);
+                            expect(reason.lookup(response.code).name).to.be(response.status);
```
Comment (czardoz) [inline test/integration-legacy/runner-spec.test.js]: This should be the other way round. Actual things should come in the `expect` call, and expected results compared with them.

Comment (kunagpal) [inline test/integration-legacy/runner-spec.test.js]: Okay, this condition will be reversed in the next commit.

File: test/integration-legacy/runner-spec.test.js [hunk 2]
```diff
@@ -251,7 +251,7 @@ describe('Runner', function () {
                             expect(cursor.ref).to.eql(runStore.ref);
 
                             expect(response.code).to.be(200);
-                            expect(reason.lookup(response.code).name).to.be(response.status);
+                            expect(response.status).to.be(reason.lookup(response.code).name);
```
Comment (shamasis) [inline test/integration-legacy/runner-spec.test.js]: This is not a valid test. You're looking up a system that itself is generating the assertion!!!

Match the status and detail with manual text.

Comment (kunagpal) [inline test/integration-legacy/runner-spec.test.js]: Wouldn't that end up being a test for `http-reasons`?

Comment (kunagpal) [inline test/integration-legacy/runner-spec.test.js]: Done

**Pull request merged**